extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -302,7 +302,7 @@ private static void buildChildren(Types.GroupBuilder builder, Iterator<SchemaEle
 
             if (schemaElement.isSetConverted_type()) {
                 final LogicalTypeAnnotation originalType = getLogicalTypeAnnotation(
-                        schemaElement.converted_type, schemaElement.logicalType, schemaElement);
+                        schemaElement.converted_type, schemaElement);
                 final LogicalTypeAnnotation newOriginalType = schemaElement.isSetLogicalType()
                         && getLogicalTypeAnnotation(schemaElement.logicalType) != null
                                 ? getLogicalTypeAnnotation(schemaElement.logicalType)
```

Comment:
We do this in two places (see L300). Should we instead do it only once? Or should we be calling `as` only once? It seems backwards to be overriding the explicit logical type with one derived from converted type.

Reply:
I try not to change the logical type + converted type inference logic because of this javadoc for logical type
```
The logical type of this SchemaElement LogicalType replaces ConvertedType, but ConvertedType is still required for some logical types to ensure forward-compatibility in format v1.
```
I am not sure where exactly converted type can be different from a non-null logical type.

But for now, I have refactored so we don't need to call `getLogicalTypeAnnotation(schemaElement.logicalType)` thrice.

Reply:
I try not to change the logical type + converted type inference logic because of this javadoc for logical type
```
The logical type of this SchemaElement LogicalType replaces ConvertedType, but ConvertedType is still required for some logical types to ensure forward-compatibility in format v1.
```
I am not sure where exactly a type inferred from a non-null converted type can be different from the type inferred from a  non-null logical type.

But for now, I have refactored so we don't need to call `getLogicalTypeAnnotation(schemaElement.logicalType)` thrice.